### PR TITLE
Add auth

### PR DESCRIPTION
**Pull request by Raigester**
Created route for user registration and JWT token generation.
Implemented functions for user creation, credential verification, and token generation.
Integrated security methods using bcrypt and JWT.

### CODE REVIEW

**Review by VladPrime11 — APPROVED** [review]

File: src/db/__init__.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from .database import Base, engine, database
-from .user import User
-from .user_room import UserRoom
-from .room_members import RoomMembers
-from .message import Message
+from .models.user import User
```
Comment (OleksandrDoronin) [inline src/db/__init__.py:2]: need to remove relative paths

File: src/db/models/message.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from sqlalchemy.orm import relationship
 from sqlalchemy.sql import func
 
-from .database import Base
+from ..database import Base
```
Comment (OleksandrDoronin) [inline src/db/models/message.py:5]: need to remove relative paths

File: src/db/models/room_members.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from sqlalchemy import Column, ForeignKey, Integer
 from sqlalchemy.orm import relationship
 
-from .database import Base
+from ..database import Base
```
Comment (OleksandrDoronin) [inline src/db/models/room_members.py:4]: need to remove relative paths

File: src/routers/auth_routers.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+router = APIRouter(
+    prefix='/auth',
+    tags=['auth']
+)
+
+db_dependency = Annotated[AsyncSession, Depends(get_db)]
+
+
+async def get_user_service(db: AsyncSession = Depends(get_db)) -> UserService:
```
Comment (OleksandrDoronin) [inline src/routers/auth_routers.py:23]: This function is called "Dependency Injection" 
I think it should be in the dependency file

File: src/routers/main_routes.py
Comment (OleksandrDoronin) [inline src/routers/main_routes.py:1]: why is this file called like that?

File: src/schemas/auth_schemas.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+class ShowUser(BaseModel):
+    id: int
+    email: EmailStr
+    username: str
+    first_name: Optional[str] = None
+    last_name: Optional[str] = None
+    date_joined: datetime
+    is_active: bool
+    is_staff: bool
+    auth0_sub: Optional[str] = None
```
Comment (OleksandrDoronin) [inline src/schemas/auth_schemas.py:25]: Do we really need to show this to the user?

File: src/service/auth_service.py
Comment (OleksandrDoronin) [inline src/service/auth_service.py:1]: Maybe we need to make a class too?

File: src/service/user_service.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        existing_user_by_username = await self.user_repository.get_user_by_username(
+            create_user_request.username
+        )
+        if existing_user_by_username:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail='Username already registered'
+            )
+
+        # Проверка существующего пользователя по email
```
Comment (OleksandrDoronin) [inline src/service/user_service.py:28]: only English

File: src/settings.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+from dotenv import load_dotenv
+from pydantic_settings import BaseSettings
+
+
+load_dotenv()
```
Comment (OleksandrDoronin) [inline src/settings.py:5]: As far as I know BaseSettings is not required load_dotenv()
https://fastapi.tiangolo.com/ru/advanced/settings/?h=basese#the-main-app-file

**Pull request merged**